### PR TITLE
fix: use timestamps to calculate apy periods for more accuracy

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

**Motivation**

https://app.shortcut.com/uma-project/story/2966/pool-apy-appears-to-be-inaccurate

**Summary**
Use timestamps to calculate more accurate periods for apy/apr calcs

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

https://app.shortcut.com/uma-project/story/2966/pool-apy-appears-to-be-inaccurate

### CODE REVIEW

File: packages/sdk/src/across/utils.ts [hunk 1]
```diff
@@ -121,33 +122,65 @@ export function percent(numerator: BigNumberish, denominator: BigNumberish): BN
...
+/**
+ * calcPeriodicCompoundInterest. Taken from https://www.calculatorsoup.com/calculators/financial/compound-interest-calculator.php?given_data=find_r&A=2&P=1&n=365&t=1&given_data_last=find_r&action=solve
+ * This will return a periodically compounded interest rate for 1 year. Multiply result by 100 for a %.
+ *
+ * @param {string} startAmount - Starting amount or price
+ * @param {string} endAmount - Ending amount or price
+ * @param {string} periodsElapsed - How many periods elapsed for the start and end amount.
+ * @param {string} periodsPerYear - How many periods in 1 year.
+ */
+export const calcPeriodicCompoundInterest = (
```
Comment (daywiss) [inline packages/sdk/src/across/utils.ts:151]: image

File: packages/sdk/src/across/clients/bridgePool.ts [hunk 2]
```diff
@@ -244,6 +269,11 @@ function joinPoolState(poolState: Awaited<ReturnType<PoolState["read"]>>): Pool
     exchangeRateCurrent: poolState.exchangeRateCurrent.toString(),
     exchangeRatePrevious: poolState.exchangeRatePrevious.toString(),
     estimatedApy,
+    estimatedApr,
+    estimatedApyBlocks,
+    estimatedAprBlocks,
```
Comment (mrice32) [inline packages/sdk/src/across/clients/bridgePool.ts:274]: Why are we still computing/returning this? Can we drop it?

File: packages/sdk/src/across/clients/bridgePool.e2e.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import get from "lodash/get";
 dotenv.config();
 
 const multicall2Address = "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696";
-const wethAddress = "0x75a29a66452C80702952bbcEDd284C8c4CF5Ab17";
+const wethAddress = "0x7355Efc63Ae731f584380a9838292c7046c1e433";
```
[the remark this answers is not in the document]
Comment (daywiss) [reply packages/sdk/src/across/clients/bridgePool.e2e.ts:13]: Had an old pool contract for weth

**Review by mrice32 — APPROVED** [review]
Looks good -- only one question about why we're still doing per-block computations?

File: packages/sdk/src/across/clients/bridgePool.ts (hunk 2 above)
Comment (daywiss) [reply packages/sdk/src/across/clients/bridgePool.ts:274]: yes i can remove

File: packages/sdk/src/across/clients/bridgePool.ts [hunk 4]
```diff
@@ -232,9 +229,25 @@ function joinUserState(
...
 }
-function joinPoolState(poolState: Awaited<ReturnType<PoolState["read"]>>): Pool {
-  const totalPoolSize = poolState.liquidReserves.add(poolState.utilizedReserves);
-  const estimatedApy = calculateApy(poolState.exchangeRateCurrent, poolState.exchangeRatePrevious);
+function joinPoolState(
+  poolState: Awaited<ReturnType<PoolState["read"]>>,
+  latestBlock: Block,
+  previousBlock: Block
+): Pool {
+  const totalPoolSize = poolState.liquidReserves.add(poolState.pendingReserves).add(poolState.utilizedReserves);
```
Comment (daywiss) [inline packages/sdk/src/across/clients/bridgePool.ts:237]: missed a regression here, bad rebase

**Pull request merged**